pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -665,6 +675,13 @@ func (s *DiscoveryServer) pushConnection(con *XdsConnection, pushEv *XdsEvent) e
 	if err := con.modelNode.SetServiceInstances(pushEv.push.Env); err != nil {
 		return err
 	}
+	if util.IsLocalityEmpty(con.modelNode.Locality) {
+		// Get the locality from the proxy's service instances.
+		// We expect all instances to have the same locality. So its enough to look at the first instance
+		if len(con.modelNode.ServiceInstances) > 0 {
+			con.modelNode.Locality = util.ConvertLocality(con.modelNode.ServiceInstances[0].GetLocality())
+		}
+	}
```

Comment:
@hzxuzhonghu May be move this check to a common method? I see the same code block in `pushConnection` and `initConnectionNode`.

Reply:
SGTM

Reply:
I tried to move it to `pilot/pkg/model/context.go` like `SetSidecarScope`. But unfortunately, pkg `pilot/pkg/model/` and `pilot/pkg/networking/util` has cycle import.